ckan/views/user.py

```diff
@@ -0,0 +1,736 @@
+    u'''Returns the URL that repoze.who will respond to and perform a
+    login or logout.'''
+    return getattr(request.environ[u'repoze.who.plugins'][u'friendlyform'],
+                   handler_name)
+
+
+def set_repoze_user(user_id, resp):
+    u'''Set the repoze.who cookie to match a given user_id'''
+    if u'repoze.who.plugins' in request.environ:
+        rememberer = request.environ[u'repoze.who.plugins'][u'friendlyform']
+        identity = {u'repoze.who.userid': user_id}
+        resp.headers.extend(rememberer.remember(request.environ, identity))
+
+
+def _edit_form_to_db_schema():
+    return schema.user_edit_form_schema()
+
+
+def _new_form_to_db_schema():
+    return schema.user_new_form_schema()
+
+
+def _extra_template_variables(context, data_dict):
+    is_sysadmin = authz.is_sysadmin(g.user)
+    try:
+        user_dict = logic.get_action(u'user_show')(context, data_dict)
+    except logic.NotFound:
+        h.flash_error(_(u'Not authorized to see this page'))
+        return
+    except logic.NotAuthorized:
+        base.abort(403, _(u'Not authorized to see this page'))
+
+    user_dict = user_dict
+    is_myself = user_dict[u'name'] == g.user
+    about_formatted = h.render_markdown(user_dict[u'about'])
+    extra = {
+        u'is_sysadmin': is_sysadmin,
+        u'user_dict': user_dict,
+        u'is_myself': is_myself,
+        u'about_formatted': about_formatted
+    }
+    return extra
+
+
+@user.before_request
+def before_request():
+    try:
+        context = dict(model=model, user=g.user, auth_user_obj=g.userobj)
+        logic.check_access(u'site_read', context)
+    except logic.NotAuthorized:
+        _, action = request.url_rule.endpoint.split(u'.')
+        if action not in (
+                u'login',
+                u'request_reset',
+                u'perform_reset',
+        ):
+            base.abort(403, _(u'Not authorized to see this page'))
+
+
+def index():
+    page_number = h.get_page_number(request.params)
+    q = request.params.get(u'q', u'')
+    order_by = request.params.get(u'order_by', u'name')
+    limit = int(
+        request.params.get(u'limit', config.get(u'ckan.user_list_limit', 20)))
+    context = {
+        u'return_query': True,
+        u'user': g.user,
+        u'auth_user_obj': g.userobj
+    }
+
+    data_dict = {
+        u'q': q,
+        u'order_by': order_by
+    }
+
+    try:
+        logic.check_access(u'user_list', context, data_dict)
+    except logic.NotAuthorized:
+        base.abort(403, _(u'Not authorized to see this page'))
+
+    users_list = logic.get_action(u'user_list')(context, data_dict)
+
+    page = h.Page(
+        collection=users_list,
+        page=page_number,
+        url=h.pager_url,
+        item_count=users_list.count(),
+        items_per_page=limit)
+
+    extra_vars = {u'page': page, u'q': q, u'order_by': order_by}
+    return base.render(u'user/list.html', extra_vars)
+
+
+# TODO: update after dashboard blueprint finished
```

Comment:
Does this still need to be done?

Reply:
No, i forgot to remove this comment after adding dashboard routes.